webapp/src/shared/components/Header/Header.tsx

```diff
@@ -0,0 +1,51 @@
+import { Link } from 'react-router-dom';
+import { Color } from '../../types';
+import { SmallAvatar } from '../Avatar/Avatar';
+import Icon, { IconSize, IconVariant } from '../Icon/Icon';
+import Status, { StatusVariant } from '../Status/Status';
+import Text, { TextColor, TextVariant, TextWeight } from '../Text/Text';
+import './Header.css';
+
+type HeaderProps = {
+  navigationFigure: string;
+  navigationUrl: string;
+  statusVariant?: StatusVariant;
+  children: string;
+};
+
+export default function Header({
+  navigationFigure,
+  navigationUrl,
+  statusVariant,
+  children,
+}: HeaderProps) {
+  return (
+    <header className="header">
+      <div className="header-navigation">
+        <Link to={navigationUrl}>
+          {navigationFigure in IconVariant ? (
+            <Icon
+              variant={navigationFigure as IconVariant}
+              size={IconSize.SMALL}
+              color={Color.LIGHT}
+            />
+          ) : (
+            <SmallAvatar url={navigationFigure} />
+          )}
+        </Link>
+      </div>
+      <div className="header-text">
+        <Text
+          variant={TextVariant.HEADING}
+          color={TextColor.LIGHT}
```

Comment:
We have been doing this so far, but I'm not sure it is a good idea.
We are hardcoding the colors. What if we want to change all the colors? Something like a night mode. Wouldn't it be better to just pass down the css values for vars like 'foreground color' and 'background color'?

Reply:
I don't understand. Could you please clarify what needs to be done? Thank you.

Reply:
So, we are hardcoding in several places these values, inside different components: `TextColor.LIGHT` and `TextColor.DARK`. I have a problem with LIGHT and DARK, because we are impliying that dark has to be dark (as close-ish to #0000000 ) and light has to be light (as close-ish to #ffffff), and I don't think we should bind ourselves to such a contract, because we may want to swap (all darks become lights, and all lights become darks).

I thought there was a problem in the dependency tree for styles, in the sense that we would set dark for some text, and then change root variables, and not have them change in our components, but I have noticed this is not happening. As things stand we have: 

```
.text-color-light {
color: var(--color-light)
} --- takes the value of `--color-light : yy` from ---->
.root {
color-dark:xx;
color-light:yy
}
```
```
.text-color-dark {
color: var(--color-dark)
} --- takes the value of `--color-dark : xx` from ---->
.root {
color-dark:xx;
color-light:yy
}
```

which is fine. If we want to change all darks, we change color-dark:xx in .root, and it works.

I think I still have a problem with dark and light. I'd rather name them like something along the lines of `light` -> `foreground`, `background` -> `background0`, and `dark` -> `background1` . But this is a concern I am just raising now, and it's outside of the scope of this PR.

Reply:
You are right. We could improve the naming of the color variables. What do we think about adding primary, secondary, tertiary, neutrals (white, black, and greys), and accent colors?
Then we could have context, like --color-primary-background, --color-secondary--background, --color-accent-submit, --color-accent-warning, --color-neutral-light, or something like that.
 Let's make a new task for this.